### PR TITLE
fix(Modal): prompt should handle defaultValue correctly

### PR DESCRIPTION
**Pull request by paranoidjk**
Fix https://github.com/ant-design/ant-design-mobile/issues/2164
Fix https://github.com/ant-design/ant-design-mobile/issues/2204

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/2221)

### CODE REVIEW

**Comment by warmhug** [comment]
问题还没有解决。看下这里 https://github.com/ant-design/ant-design-mobile/commit/0d2981dfe4da73a0118f3da98c2b055eaa075253#commitcomment-26734191

**Comment by warmhug** [comment]
#2204 这个也应该是有问题的，一起看看

**Comment by paranoidjk** [comment]
@warmhug 好了

**Comment by paranoidjk** [comment]
@silentcloud  @zhang740 也可以看下，这里比较特殊，Modal 内部的 Input 不会提供 onChange 给用户，只有一个类似 onConfirm, onSubmit 的最终回调

File: components/modal/prompt.tsx [hunk 1]
```diff
@@ -18,7 +17,9 @@ export default function prompt(
 
   const prefixCls = 'am-modal';
 
-  let data: any = {};
+  let data: any = {
+    text: defaultValue,
```
Comment (paranoidjk) [inline components/modal/prompt.tsx]: readme API 文档 里面写 password 不支持设置 defaultValue 的设计小版本就不改了，以后如果有用户有需求再看

**Comment by silentcloud** [comment]
我记得之前 https://github.com/ant-design/ant-design-mobile/issues/2204 是修过的，怎么又报出来了

**Comment by paranoidjk** [comment]
这两个 bug 都是同一类问题，没处理好 defaultValue => value => onChange(value) 的处理

**Pull request merged**